src/Tribe/Views/V2/Views/Day_View.php

```diff
@@ -41,30 +41,22 @@ public function prev_url( $canonical = false, array $passthru_vars = [] ) {
 		$date           = $this->context->get( 'event_date', $default_date );
 		$event_date_var = $default_date === $date ? '' : $date;
 
-		$previous_date = date( Dates::DBDATEFORMAT, strtotime( $date . '00:00 -1day' ) );
-
-		$past = tribe_events()->by_args( [
-			'on_date'  => $previous_date,
-			'posts_per_page' => 1,
-			'order'          => 'DESC',
-		] );
-		$past_posts = $past->get_query()->get_posts();
-		$previous_event = reset( $past_posts );
-
-		if ( ! $previous_event ) {
-			$past = tribe_events()->by_args( [
-				'starts_before'  => $date,
-				'posts_per_page' => 1,
-				'order'          => 'DESC',
-			] );
-			$past_posts = $past->get_query()->get_posts();
-			$previous_event = reset( $past_posts );
-
-			if ( ! $previous_event ) {
+		$previous_date = date( Dates::DBDATEFORMAT, strtotime( $date . ' -1day' ) );
+
+		$prev = tribe_events()->where( 'date_overlaps', tribe_beginning_of_day( $previous_date ), tribe_end_of_day( $previous_date ) )->order( 'DESC' )->per_page( 1 );
```

Comment:
Using a "fresh" call to `tribe_events` here will invalidate the repository arguments set in `Day_View::setup_repository_args`.
Change the code to this to keep the arguments:

```php
$prev_args = $this->setup_repository_args( $this->context );
unset( $prev_args['order'] );
$prev_args[ 'date_overlaps' ] = [ tribe_beginning_of_day( $previous_date ), tribe_end_of_day( $previous_date ) ];
$prev = tribe_events()->by_args( $prev_args )->per_page( 1 );
```